test/battle/ability/commander.c

```diff
@@ -0,0 +1,420 @@
+    }
+}
+
+DOUBLE_BATTLE_TEST("(Commander) Attacker is kept (Dondozo Left Slot)")
+{
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_SURF].target == MOVE_TARGET_FOES_AND_ALLY);
+        PLAYER(SPECIES_WOBBUFFET);
+        PLAYER(SPECIES_TATSUGIRI) { Ability(ABILITY_COMMANDER); }
+        PLAYER(SPECIES_DONDOZO);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponentRight, MOVE_TACKLE, target: opponentLeft); }
+        TURN { SWITCH(playerLeft, 2); MOVE(opponentLeft, MOVE_SURF); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, opponentRight);
+        ABILITY_POPUP(playerRight, ABILITY_COMMANDER);
+        MESSAGE("Tatsugiri was swallowed by Dondozo and became Dondozo's commander!");
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_SURF, opponentLeft);
+        HP_BAR(playerLeft);
+        MESSAGE("Foe Wobbuffet's attack missed!");
+        HP_BAR(opponentRight);
+    }
+}
+
+DOUBLE_BATTLE_TEST("(Commander) Attacker is kept (Dondozo Right Slot)")
+{
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_SURF].target == MOVE_TARGET_FOES_AND_ALLY);
+        PLAYER(SPECIES_TATSUGIRI) { Ability(ABILITY_COMMANDER); }
+        PLAYER(SPECIES_WOBBUFFET);
+        PLAYER(SPECIES_DONDOZO);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponentRight, MOVE_TACKLE, target: opponentLeft); }
+        TURN { SWITCH(playerRight, 2); MOVE(opponentLeft, MOVE_SURF); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, opponentRight);
+        ABILITY_POPUP(playerLeft, ABILITY_COMMANDER);
+        MESSAGE("Tatsugiri was swallowed by Dondozo and became Dondozo's commander!");
+        MESSAGE("Foe Wobbuffet's attack missed!");
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_SURF, opponentLeft);
+        HP_BAR(playerRight);
+        HP_BAR(opponentRight);
+    }
+}
+
+DOUBLE_BATTLE_TEST("(Commander) Tatsugiri does not attack if Dondozo faints the same turn it's switched in")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        PLAYER(SPECIES_TATSUGIRI) { Ability(ABILITY_COMMANDER); }
+        PLAYER(SPECIES_DONDOZO) { HP(1); }
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN {
+            SWITCH(playerLeft, 2);
+            MOVE(opponentLeft, MOVE_TACKLE, target: playerLeft);
+            MOVE(opponentRight, MOVE_TACKLE, target: playerRight);
+            MOVE(playerRight, MOVE_CELEBRATE);
+            SEND_OUT(playerLeft, 0);
+        }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, opponentLeft);
+        HP_BAR(playerLeft);
+        MESSAGE("Dondozo fainted!");
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, opponentRight);
+        HP_BAR(playerRight);
+        NOT MESSAGE("Tatsugiri used Celebrate!");
+    }
+}
+
+DOUBLE_BATTLE_TEST("(Commander) Tatsugiri gets hit by Dragon Darts when a commanded Dondozo faints")
```

Comment:
```suggestion
DOUBLE_BATTLE_TEST("Commander Tatsugiri gets hit by Dragon Darts when a commanded Dondozo faints")
```

Reply:
Changed the text to match the intended behavior.